File: user-service/src/main/java/rs/raf/user_service/domain/dto/CreateCompanyDto.java
package rs.raf.user_service.domain.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;


@Getter
@Setter
@NoArgsConstructor
public class CreateCompanyDto {
    @NotNull(message = "Name must not be null")
    @NotBlank
    private String name;
    @NotNull(message = "registrationNumber must not be null")
    @NotBlank
    private String registrationNumber;
    @NotNull(message = "taxId must not be null")
    @NotBlank
    private String taxId;
    @NotNull(message = "activityCode must not be null")
    @NotBlank
    private String activityCode;
    @NotNull(message = "address must not be null")
    @NotBlank
    private String address;
    @NotNull(message = "majorityOwner must not be null")
    @Min(1L)
    private Long majorityOwner;
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/dto/company/CreateCompanyDto.kt
package pack.userservicekotlin.domain.dto.company

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class CreateCompanyDto(
    @NotNull(message = "Name must not be null")
    @NotBlank
    val name: String? = null,
    @NotNull(message = "registrationNumber must not be null")
    @NotBlank
    val registrationNumber: String? = null,
    @NotNull(message = "taxId must not be null")
    @NotBlank
    val taxId: String? = null,
    @NotNull(message = "activityCode must not be null")
    @NotBlank
    val activityCode: String? = null,
    @NotNull(message = "address must not be null")
    @NotBlank
    val address: String? = null,
    @NotNull(message = "majorityOwner must not be null")
    val majorityOwner:
        @Min(1L)
        Long? = null,
)
